crates/web5/src/credentials/status_list_credential.rs

```diff
@@ -0,0 +1,560 @@
+use std::collections::HashMap;
+use std::io::{Read, Write};
+
+use super::verifiable_credential_1_1::{VerifiableCredential, VerifiableCredentialCreateOptions};
+use crate::credentials::{CredentialSubject, Issuer};
+use crate::errors::{Result, Web5Error};
+use crate::json::{JsonObject, JsonValue};
+use base64::Engine;
+use flate2::{read::GzDecoder, write::GzEncoder, Compression};
+
+pub const STATUS_LIST_CREDENTIAL_CONTEXT: &str = "https://w3id.org/vc/status-list/2021/v1";
+pub const STATUS_LIST_CREDENTIAL_TYPE: &str = "StatusList2021Credential";
+pub const STATUS_LIST_2021: &str = "StatusList2021";
+
+pub const STATUS_LIST_2021_ENTRY: &str = "StatusList2021Entry";
+
+pub struct StatusListCredential {
+    pub base: VerifiableCredential,
+}
+
+impl StatusListCredential {
+    pub fn create(
+        issuer: Issuer,
+        status_purpose: String,
+        disabled_credentials: Option<Vec<VerifiableCredential>>,
+    ) -> Result<Self> {
+        // Determine the status list indexes based on the provided credentials to disable.
+        let status_list_indexes = match disabled_credentials {
+            Some(credentials) => Self::get_status_list_indexes(&status_purpose, credentials)?,
+            None => Vec::new(),
+        };
+
+        // Generate the base64 bitstring from the status list indexes.
+        let base64_bitstring = Self::bitstring_generation(status_list_indexes)?;
+
+        // Construct the properties for the credential subject.
+        let additional_properties = {
+            let mut properties = HashMap::new();
+            properties.insert(
+                "statusPurpose".to_string(),
+                JsonValue::String(status_purpose.clone()),
+            );
+            properties.insert(
+                "type".to_string(),
+                JsonValue::String(STATUS_LIST_2021.to_string()),
+            );
+            properties.insert(
+                "encodedList".to_string(),
+                JsonValue::String(base64_bitstring.clone()),
+            );
+            JsonObject { properties }
+        };
```

Comment:
Easier to see that we're just initting a hashmap with a few values. You can also use `HashMap::from()` if you prefer, instead of `into_iter().collect()`.
```suggestion
        let additional_properties = JsonObject {
            properties: [
                ("statusPurpose".to_string(), JsonValue::String(status_purpose)),
                ("type".to_string(), JsonValue::String(STATUS_LIST_2021.to_string())),
                ("encodedList".to_string(), JsonValue::String(base64_bitstring)),
            ].into_iter().collect(),
        };
```

Reply:
genius.